ddtrace/internal/writer.py

```diff
@@ -429,10 +431,16 @@ def flush_queue(self, raise_exc=False):
                     # it's not thread-safe.
                     # https://github.com/DataDog/datadogpy/issues/439
                     # This really isn't ideal as now we're going to do a ton of socket calls.
-                    self.dogstatsd.distribution("datadog.tracer.http.sent.bytes", len(encoded))
-                    self.dogstatsd.distribution("datadog.tracer.http.sent.traces", len(enc_traces))
+                    self.dogstatsd.distribution(  # type: ignore[union-attr]
```

Comment:
why is this ignore needed here?

Reply:
or for all of the dogstatsd calls?

Reply:
For `writer`, the initializer accepts `dogstatsd` as an optional arg so mypy doesn't like that `self.dogstatsd` which can be `None` calls `distribution()`.

Reply:
in that case, should we instead do:

```python
if self.report_metrics and self.dogstatsd:
    self.dogstatsd....
```

mypy is pretty good about honoring our checking.

Reply:
This makes sense, my only concern is if there's any cases where a user would ever call `flush_queue()` without setting a `dogstatsd` attribute to the agent?

Reply:
if it was, then we wouldn't have it as `Optional`... until we can remove the `Optional`, then we'll be better off having the none check here.